server/graphql/v2/mutation/TransactionMutations.ts

```diff
@@ -2,16 +2,59 @@ import express from 'express';
 import { GraphQLNonNull, GraphQLString } from 'graphql';
 
 import orderStatus from '../../../constants/order_status';
-import { canReject } from '../../../graphql/common/transactions';
 import { purgeCacheForCollective } from '../../../lib/cache';
 import { notifyAdminsOfCollective } from '../../../lib/notifications';
 import models from '../../../models';
+import { canReject, createPlatformTipTransaction } from '../../common/transactions';
 import { Forbidden, NotFound, Unauthorized } from '../../errors';
 import { refundTransaction as legacyRefundTransaction } from '../../v1/mutations/orders';
+import { AmountInput } from '../input/AmountInput';
 import { fetchTransactionWithReference, TransactionReferenceInput } from '../input/TransactionReferenceInput';
 import { Transaction } from '../interface/Transaction';
 
 const transactionMutations = {
+  addPlatformTipToTransaction: {
+    type: new GraphQLNonNull(Transaction),
+    description: 'Add platform tips to a transaction',
+    args: {
+      transaction: {
+        type: new GraphQLNonNull(TransactionReferenceInput),
+        description: 'Reference to the transaction in the platform tip',
+      },
+      amount: {
+        type: GraphQLNonNull(AmountInput),
+        description: 'Reference to the amount of input for the platform tip',
+      },
+    },
+    async resolve(_: void, args, req: express.Request): Promise<typeof Transaction> {
+      if (!req.remoteUser) {
+        throw new Unauthorized('You need to be logged in to add a platform tip');
+      }
+
+      const transaction = await fetchTransactionWithReference(args.transaction);
+
+      if (!req.remoteUser.isAdmin(transaction.HostCollectiveId)) {
+        throw new Unauthorized('Only host admins can add platform tips');
+      }
+
+      const isPlatformTipAvailable = await models.Transaction.findOne({
+        where: {
+          TransactionGroup: transaction.TransactionGroup,
+          kind: 'PLATFORM_TIP',
+        },
+      });
+
+      if (isPlatformTipAvailable) {
+        throw new Error('Platform tip is already available for this transaction group');
+      }
+
+      const payload = {
+        transaction,
+        amount: args.amount.valueInCents,
+      };
+      return await createPlatformTipTransaction(payload);
```

Comment:
Let's rather use https://github.com/opencollective/opencollective-api/blob/main/server/models/Transaction.js#L567. You'll probably need to update the transaction with `isFeesOnTop` + `platformFeeInHostCurrency` first.

Reply:
Note: we'll also need to update the opposite transaction (CREDIT/DEBIT)

Reply:
@Betree : Done. 😄 So my thinking is opposite transaction is auto updated within https://github.com/opencollective/opencollective-api/blob/60088ad8718da967b78491b7f2f4a25d946d3672/server/models/Transaction.js#L608 ? Sorry, maybe I am confused by this function. 🤔 

On another note, so I've got rid of my custom function `createPlatformTipTransaction` and simplified the logic at, https://github.com/opencollective/opencollective-api/pull/5402/commits/5de069717a2d3a6c7fa033771884c4f179b74393#diff-37b360415f67c48408202a4754178f4bd2d792651489f40714fc6cff77e04a65R55-R66. Let me know if that looks good. 😃